module/zfs/zfs_vnops.c

```diff
@@ -278,6 +278,10 @@ zfs_holey_common(struct inode *ip, int cmd, loff_t *off)
 	if (error == ESRCH)
 		return (SET_ERROR(ENXIO));
 
+	/* file was dirty, so fall back to using file_sz logic */
+	if (error == EBUSY)
```

Comment:
Yes, upon further reflection I'm OK with that and withdraw my objection.  The existing patch leaves this check where it is.